### ISSUE TITLE
[Feature Request]: Jpeg xl

### ISSUE THREAD

**Issue by Liverel** [ollm/OpenComic#165]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/ollm/OpenComic/issues) for a feature request that matches the one I want to file, without success.

### Describe your suggested feature

Support the Jpeg xl format.

**Comment by ollm** [comment]
At the moment I am waiting to see if Chromium adds JPEG XL support again, since this would make the implementation much easier (Only the extension would have to be recognized as a compatible format).

https://bugs.chromium.org/p/chromium/issues/detail?id=1451807
https://bugs.chromium.org/p/chromium/issues/detail?id=1178058

**Comment by Reireiyan** [comment]
Ok, so this basically means everything that is based on chromium is going to just not have JXL support. Good job, chromium developers. I really hope they make the right choice down the road and actually support this format.

**Comment by ollm** [comment]
I'm happy to say that I've added preliminary support for `JPEG XL` bd6fb0a8c8be3685b48c88fa21bbe2d3c384a491 via the [`jxl-oxide-wasm`](https://www.npmjs.com/package/jxl-oxide-wasm) library, which supports most of the features of `JPEG XL`.

If you want to try it to see if there are any bugs, I'll leave you a build of the master branch.

Buid: https://mega.nz/file/WGBimRDT#Nt3y0E5GofDJZvYukH0XxaERgms4xJRo0hhU3Iq8QO4

**Comment by Liverel** [comment]
YES! This is amazing. The JPEG XL images work now. Oh, and I ended up building from master myself since my system is Linux; I can't test the exe.

**Comment by woducku** [comment]
![Image](https://github.com/user-attachments/assets/f088a85e-82e9-4b5c-8843-0efff925238c)

works pretty nice. few readers mess up when the source is lossy. is there any eta for official release?

**Comment by ollm** [comment]
> is there any eta for official release?

I don't have a specific date for the next release, I still plan to add some more requested features, such as OPDS.

Although I can say that apart from OPDS I don't think it will add much more, since there are already multiple new functions implemented: [v1.5.0 Changelog](https://github.com/ollm/OpenComic/blob/be5cd981ae26476862cb59281bebdf1f61e8681a/CHANGELOG.md#v150-dd-mm-yyyy)

**Comment by ollm** [comment]
Closing this, since it's now included in [`v1.6.0`](https://opencomic.app/docs/installation/download)

**Comment by YRSB** [comment]
Hi! Google Chrome 145 released with JPEG-XL image support.

**Comment by ollm** [comment]
Thanks for the update!

For now, I'll continue with the current implementation using [`jxl-oxide-wasm`](https://www.npmjs.com/package/jxl-oxide-wasm). Once Electron ships with Chromium 145 and Sharp includes it in its precompiled binaries https://github.com/lovell/sharp/issues/2731#issuecomment-3643368319, I'll switch from the current approach to the native Chrome/Sharp implementation.
